wsg_50_driver/include/wsg_50/optical_weiss_finger.h

```diff
@@ -0,0 +1,341 @@
+
+    /**
+     * @brief OpticalWeissFinger: Constructor
+     * @param nh: Node handle for grabbing params
+     * @param param_prefix: Prefix for param keys
+    */
+    OpticalWeissFinger(ros::NodeHandle& nh, std::string param_prefix);
+
+    /**
+     * @brief ~FMFWeissFinger: Destructor
+    */
+    ~OpticalWeissFinger();
+
+    /**
+     * @brief do_calibration: Performs calibration
+     * @param cmd_mutex: Need to acquire this mutex before sending commands to hand
+     * @return Whether or not the calibration was successful
+    */
+    bool do_calibration(std::mutex* cmd_mutex);
+
+    /**
+     * @brief load_calibration: Load calibration, including sending params to hand if necessary
+     * @param cmd_mutex: Need to acquire this mutex before sending commands to hand
+     * @return Whether or not the calibration was successfully loaded
+    */
+    bool load_calibration(std::mutex* cmd_mutex);
+
+    /**
+     * @brief start_reading: Initializes and begins reading from this finger
+     * @param cmd_mutex: Need to acquire this mutex before sending commands to hand
+     * @return Whether or not reading was successfully started
+    */
+    bool start_reading(std::mutex* cmd_mutex);
+
+    /**
+     * @brief pause_reading: Stop reading data from this finger. Should only be called
+     * after start_reading()
+     * @return Whether or not reading was successfully paused
+    */
+    bool pause_reading();
+
+    /**
+     * @brief restart_reading: Resume reading data from this finger. Should only be
+     * called after pause_reading().
+     * @return  Whether or not not reading was successfully resumed.
+    */
+    bool restart_reading();
+
+    /**
+     * @brief clear_sample_buffer: Remove previous samples from the sample buffer
+     * @return Whether or not the sample buffer was successfully cleared
+    */
+    bool clear_sample_buffer();
+
+    /**
+     * @brief get_sample: Get the latest data sample
+     * @return The latest data sample
+    */
+    wsg_50_common::WeissFingerData get_sample();
+
+    /**
+     * @brief get_latest_samples: Get the most recent n samples, ordered from newest to oldest
+     * @param n_msgs: The number of samples to get. Will return all samples if less than or equal to zero
+     * @param buff: Container for the n samples
+     * @return True if at least one sample was returned, otherwise False
+    */
+    bool get_latest_samples(unsigned int n_msgs, std::vector<wsg_50_common::WeissFingerData>& buff);
+
+    /**
+     * @brief hand_width_to_finger_tip_separation: Computes the distance from this finger to the target on
+     *        the other finger.
+     * @param width: The current width of the hand
+     * @return The distance between the surface of this finger and the target on the other fingertip
+    */
+    double hand_width_to_finger_tip_separation(double width);
+private:
+
+    ros::NodeHandle nh_; // Node handle used to create publisher of data
```

Comment:
We should strip this out- I think we want to leave all the `ros` stuff for another library.

Reply:
Another thing too is that we should have the data originally published by `ros` nodes here easily accessible from methods that whatever `ros` layer we write can call.